### PR TITLE
Stop supporting android versions less than Lollipop

### LINKED ISSUE #2260: Stop supporting Android version < 5.0 (Lollipop)
These older versions of Android won't be able to read new Zimit ZIM files relying on ServiceWorkers.

On the top of this, these Android OSes are more than 6 years old and require more and more effort to developers to keep them working fine.

My personal experience is that many of these devices are - from a hardware perspective - so limited that this is really difficult to provide a good UX with with recent versions of Kiwix Android.

Google play reports that stopping supporting these older versiona of Android would impact around 4.000 users which would still be able to download Kiwix Android 3.3.4 but not get any updated/newer version of Kiwix Android.

I plan to keep this ticket open to comment for 1-2 weeks to gather feedbacks. Then a decision would be made for future 3.4 version.

### PR DESCRIPTION
**Pull request by abdulwd**
Fixes #2260 

- Increment the minSdkVersion to 21
- Remove obsolete min version checks
- Fixed some lint errors
- Ignoring ConvertToWebp lint warning for now
- Removed `app:srcCompat`

### CODE REVIEW

**Comment by Frans-Lukas** [comment]
Is the lollipop check necessary as well? 

`ServiceWorkerInitialiser.kt:39`
```kotlin

  override fun shouldInterceptRequest(request: WebResourceRequest): WebResourceResponse? {
    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
      zimReaderContainer.load(request.url.toString())
    } else {
      null
    }
  }

```

**Comment by abdulwd** [comment]
I'll remove this block. The linked issue has been fixed.

**Comment by macgills** [comment]
```
  /home/runner/work/kiwix-android/kiwix-android/core/src/main/res/layout/activity_page.xml:18: Error: Use SwitchCompat from AppCompat or SwitchMaterial from Material library [UseSwitchCompatOrMaterialXml]
      <Switch
      ^
  /home/runner/work/kiwix-android/kiwix-android/core/src/main/res/layout/item_help.xml:24: Error: Must use app:tint instead of android:tint [UseAppTint]
      android:tint="?colorOnSurface"
```

static analysis failures. Also were we going to replace the `srcCompat` or is that a another ticket do you think?

**Comment by abdulwd** [comment]
I'll fix these failures. `srcCompat` as well I'll do in this PR.

File: core/src/main/java/org/kiwix/kiwixmobile/core/base/BaseActivity.kt [hunk 1]
```diff
@@ -50,16 +48,6 @@ abstract class BaseActivity : AppCompatActivity() {
     unbinder = ButterKnife.bind(this)
   }
 
-  // TODO https://issuetracker.google.com/issues/141132133 remove this once appcompat has been fixed
-  override fun applyOverrideConfiguration(overrideConfiguration: Configuration?) {
-    if (Build.VERSION.SDK_INT in Build.VERSION_CODES.LOLLIPOP..Build.VERSION_CODES.N_MR1 &&
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/base/BaseActivity.kt:55]: Oh was this bug fixed?

Comment (abdulwd) [reply core/src/main/java/org/kiwix/kiwixmobile/core/base/BaseActivity.kt:55]: Apparently, the link says it's closed. I had to bump the version of `appcompat` as the tests were failing which I reckon was due to this issue.

Comment (macgills) [reply core/src/main/java/org/kiwix/kiwixmobile/core/base/BaseActivity.kt:55]: well groovy

File: app/src/main/res/layout/item_download.xml [hunk 2]
```diff
@@ -1,106 +1,103 @@
...
-  android:orientation="horizontal"
-  android:paddingStart="@dimen/activity_horizontal_margin"
-  android:paddingTop="@dimen/activity_vertical_margin"
-  android:paddingEnd="@dimen/activity_horizontal_margin">
-
-  <ImageView
-    android:id="@+id/libraryDownloadFavicon"
-    android:layout_width="@dimen/favicon_width"
+  <LinearLayout xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
```
Comment (Frans-Lukas) [inline app/src/main/res/layout/item_download.xml:8]: Just general clean-up or part of the Lollipop change?

Comment (abdulwd) [reply app/src/main/res/layout/item_download.xml:8]: Hide white space changes before reviewing.
This was just fixing incorrect indentations.

Comment (Frans-Lukas) [reply app/src/main/res/layout/item_download.xml:8]: Didn't know that existed. Cool!

Comment (Frans-Lukas) [reply app/src/main/res/layout/item_download.xml:8]: In that case this looks good!

**Review by Frans-Lukas — APPROVED** [review]

**Comment by macgills** [comment]
I think there is a line in CoreApp that enables vector support too

**Review by macgills — APPROVED** [review]

**Comment by macgills** [comment]
Great stuff, ahhh so happy all this noisy code is gone

**Pull request merged**